lib/bolt/transport/base.rb

```diff
@@ -47,7 +47,10 @@ def with_events(target, callback)
         callback.call(type: :node_result, result: result) if callback
         result
       rescue StandardError => ex
-        Bolt::Result.from_exception(target, ex)
+        result = Bolt::Result.from_exception(target, ex)
+        @logger.debug("Failure on #{target.uri}: #{JSON.dump(result.value)}")
+        callback.call(type: :node_result, result: result) if callback
```

Comment:
this will double report if callback.call fails. Is it worth guarding against bugs with an extra begin/rescue around that?

Reply:
If `callback.call` fails, who really knows what will be reported? That shouldn't really be able to fail aside from a bug in bolt though, so it doesn't seem too concerning.

Reply:
I think tests will do a better job guarding against those bugs. I'd be tempted to change this control-flow a bit though: if there's an exception in the yield, we should catch it and wrap it in Bolt::Result. Does it really make sense to do the same for an exception in the callback?

Reply:
I added a 2nd commit simplifying this block and intentionally only catching errors from the `yield`.

Reply:
I believe we create empty results somewhere else and a result with a callback traceback is propabably about as useful

Reply:
I'm not sure what your answer implies. Not really familiar with where and why we create empty results.